mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -21,9 +26,59 @@ public AnnotationHandlerMapping(final Object... basePackage) {
 
     public void initialize() {
         log.info("Initialized AnnotationHandlerMapping!");
+        for (var pack : basePackage) {
+            registerControllersInPackage(pack);
+        }
+    }
+
+    private void registerControllersInPackage(Object pack) {
+        Reflections reflections = new Reflections(pack.toString());
+        Set<Class<?>> controllers = reflections.getTypesAnnotatedWith(Controller.class);
+
+        for (Class<?> controller : controllers) {
+            Object controllerInstance = instantiateController(controller);
+            Method[] methods = controller.getDeclaredMethods();
+            registerHandlerMethod(methods, controllerInstance);
+        }
     }
 
-    public Object getHandler(final HttpServletRequest request) {
-        return null;
+    private void registerHandlerMethod(Method[] methods, Object controllerInstance) {
+        for (Method method : methods) {
+            if (method.isAnnotationPresent(RequestMapping.class)) {
+                RequestMapping requestMapping = method.getAnnotation(RequestMapping.class);
+                String url = requestMapping.value();
+                RequestMethod[] requestMethods = requestMapping.method();
+
+                for (var requestMethod : requestMethods) {
+                    HandlerKey handlerKey = new HandlerKey(url, requestMethod);
+                    HandlerExecution handlerExecution = new HandlerExecution(controllerInstance, method);
+
+                    handlerExecutions.put(handlerKey, handlerExecution);
+                }
+            }
+        }
+    }
+
+    private Object instantiateController(Class<?> controller) {
+        Object controllerInstance;
+        try {
+            controllerInstance = controller.getConstructor().newInstance();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("excetion occured" + e.getMessage());
```

Comment:
이 부분은 현재 크게 문제가 되는 부분은 아니지만, 정확히 왜 error가 발생했는지 원인을 더 좁혀서 작성해주면 좋을 것 같아요~ 
e.getMessage()를 보면 파악할 수는 있겠지만, message만 봐서는 헷갈릴 것 같아서요ㅠ 미래의 슬링키를 위해서 해보는 거 어떠신가용

```suggestion
} catch (Exception e) {
                throw new IllegalStateException(
    "Failed to instantiate controller: " + controller.getName(), e);
```

이런식으로 `현재 상황에 대한 설명`, `어떤 변수의 값 때문에 발생했는지`를 남겨주면 좋을 것 같아요~ 이 부분 말고 다른 에러로그도 마찬가지로 해놓으면 나중에 편하지 않을까요! 
물론 슬링키만의 확고한 이유가 있으면 괜찮을 것 같아요!

Reply:
> 미래의 슬링키를 위해서 해보는 거 어떠신가용

이 부분 공감되네요 🤣 에러 로그 부분에서 꼼꼼하게 확인하지 않았던 거 같아요. 한번 적용해볼게요!